### PR TITLE
Ci fixes - Part II

### PR DESCRIPTION
**Pull request by stackmystack**
This branch has fixes for the CI pipeline:

1. Making MSSQL unit tests go through
2. Making PostgreSQL and MySQL properly execute

### CODE REVIEW

File: lib/arel_extensions.rb [hunk 1]
```diff
@@ -77,7 +77,13 @@ class Arel::Nodes::Case
 require 'arel_extensions/nodes/cast'
 require 'arel_extensions/nodes/json'
 
-
+if defined?(Arel::Visitors::SQLServer)
```
Comment (akimd) [inline lib/arel_extensions.rb:80]: Please, put comments documenting dark magic.  Commit messages are not enough.  Especially not for obscure stuff.

File: test/arelx_test_helper.rb [hunk 2]
```diff
@@ -6,6 +6,58 @@
 
 require 'support/fake_record'
 
+$colored = ENV["TERM"] =~ /^xterm|-256color$/ ? ->(color, msg){ "\x1b[#{color}m#{msg}\x1b[89m\x1b[0m" } : ->(color, msg){ "#{msg}" }
```
Comment (akimd) [inline test/arelx_test_helper.rb:9]: Why a global lambda instead of merely a function?

File: test/arelx_test_helper.rb [hunk 3]
```diff
@@ -6,6 +6,58 @@
...
+def load_lib(gem)
+  begin
+    Gem::Specification.find_by_name(gem)
+    require gem
+  rescue Gem::MissingSpecError
+    warn "Warning: failed to load gem #{gem}. Are you sure it's installed?"
+  end
+end
+
+dbs_to_test.map { |_, gem| load_lib(gem)}
```
Comment (akimd) [inline test/arelx_test_helper.rb:59]: Looks like an `each` to me.  You could inline load_lib in it.

File: test/with_ar/all_agnostic_test.rb [hunk 4]
```diff
@@ -171,7 +171,7 @@ def test_aggregation_with_ar_calculation
         # This should works no matter which version of rails is used
         assert User.group(:score).average(:id).values.all?{|e| !e.nil?}
 
-        # Since Rails 7, a patch to calculations.rb has tirggered a double 
+        # Since Rails 7, a patch to calculations.rb has tirggered a double
```
Comment (akimd) [inline test/with_ar/all_agnostic_test.rb:174]: While at it, please fix the "tirgerred".

File: Rakefile [hunk 5]
```diff
@@ -23,6 +23,8 @@ namespace :test do
     t.verbose = true
     t.ruby_opts = ["--dev"] if defined?(JRUBY_VERSION)
   }
+
+  ENV['DB'] = 'none'
```
Comment (akimd) [inline Rakefile:27]: Please write the commit message here.

**Review by akimd — COMMENTED** [review]
LGTM.
Your commit messages are great.  But please, make some of them comments, to make it easier to understand the code.

File: lib/arel_extensions.rb (hunk 1 above)
Comment (stackmystack) [reply lib/arel_extensions.rb:80]: Done.

File: test/arelx_test_helper.rb (hunk 2 above)
Comment (stackmystack) [reply test/arelx_test_helper.rb:9]: No particular reason other than avoiding querying ENV over and  over. It's silly, so I'll convert it to a func.

File: test/arelx_test_helper.rb (hunk 3 above)
Comment (stackmystack) [reply test/arelx_test_helper.rb:59]: Actually this whole thing was a vestigial remain of something else. I am redoing it to something that makes sense.

File: Rakefile (hunk 5 above)
Comment (stackmystack) [reply Rakefile:27]: It's going away. Pointless.

**Pull request merged**